datafusion/common/src/scalar.rs

```diff
@@ -170,6 +172,7 @@ impl PartialEq for ScalarValue {
             (IntervalMonthDayNano(_), _) => false,
             (Struct(v1, t1), Struct(v2, t2)) => v1.eq(v2) && t1.eq(t2),
             (Struct(_, _), _) => false,
+            (Null, _) => false,
```

Comment:
I think we need to also add the postive case:

```suggestion
            (Null, Null) => true,
            (Null, _) => false,
```